agent/hook.go

```diff
@@ -50,6 +51,19 @@ func (a *Agent) ExecuteHooks(ctx context.Context, request *corev2.CheckRequest,
 	return executedHooks
 }
 
+// errorHookConfig will return a hookConfig that will echo its error argument
+// and exit status 2.
+func errorHookConfig(namespace, name string, err error) *corev2.HookConfig {
+	return &corev2.HookConfig{
+		ObjectMeta: corev2.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Command: fmt.Sprintf("echo 'hook %s: %s'; exit 2", name, err),
```

Comment:
The resulting command was a bit confusing to me in my tests, e.g.
```
hooks:
    - command: 'echo ''hook prout: missing hook config''; exit 2'
```

Is this the intended behaviour?

Reply:
That looks like yaml serialization weirdness to me. I'm not sure if it's a problem or not...

Reply:
According to the yaml specification, two single quotes are how single quotes are escaped in yaml. https://yaml.org/spec/current.html#id2534365